docs/developer/architecture/sso.rst

```diff
@@ -0,0 +1,126 @@
+Single Sign On
+==============
+
+Context
+-------
+
+As for now, when we log in into the PlatformUI then we click on grafana,
+we have to log in again because these two components do not share a common
+session and Dex (the authentication provider) does not offer such
+fonctionality.
+
+Requirements
+------------
+* Ability to share login session between UI components
+* Ability to logout once and for all UI components
+
+User Stories
+------------
+As an application administrator I want to:
+
+- While logged in, if I click on "advanced metrics" I want to access Grafana
+  without logging in again
+- While logged in, If I click logout I want to be logged out from all other
+  components
+- This functionality should be available in both static password and other
+  authentication providers
+
+Design Choices
+--------------
+
+Rejected Designs
+****************
+
+Custom Keycloak
+~~~~~~~~~~~~~~~
+Keycloak does persist a session and thus can support an SSO behavior, But using
+it means maintaining a database while we want a simple authentication provider
+to fallback on.
+
+Deprecate Dex
+~~~~~~~~~~~~~
+The issue of SSO is that Dex does not keep track of the session, we may as well
+get rid of Dex and replace it with other mechanism.
+The issue with this solution is that the replacement, oauth2-proxy, does not
+provide oidc functionalities and thus cannot be used by the kube-apiserver.
```

Comment:
Maybe we should detail why authenticating using a proxy on kube-apiserver (https://kubernetes.io/docs/reference/access-authn-authz/authentication/#authenticating-proxy) is not an option we retained